logicrunner/handle_executorresults.go

```diff
@@ -0,0 +1,201 @@
+			es.PendingConfirmed = false
+		} else if p.msg.Pending == message.NotPending {
+			logger.Debug("executor we came to thinks that execution pending, but previous said to continue")
+
+			es.pending = message.NotPending
+		}
+	} else if es.pending == message.PendingUnknown {
+		es.pending = p.msg.Pending
+
+		if es.pending == message.PendingUnknown {
+			p.Result.clarifyPending = true
+		}
+	}
+
+	// set false to true is good, set true to false may be wrong, better make unnecessary call
+	if !es.LedgerHasMoreRequests && p.msg.LedgerHasMoreRequests {
+		es.LedgerHasMoreRequests = p.msg.LedgerHasMoreRequests
+	}
+
+	// prepare Queue
+	if p.msg.Queue != nil {
+		queueFromMessage := make([]ExecutionQueueElement, 0)
+		for _, qe := range p.msg.Queue {
+			queueFromMessage = append(
+				queueFromMessage,
+				ExecutionQueueElement{
+					ctx:     qe.Parcel.Context(context.Background()),
+					parcel:  qe.Parcel,
+					request: qe.Request,
+				})
+		}
+		es.Queue = append(queueFromMessage, es.Queue...)
+	}
+
+	es.Unlock()
+
+	return nil
+}
+
+// ---------------
+
+type HandleExecutorResults struct {
+	dep *Dependencies
+
+	Message bus.Message
+}
+
+func (h *HandleExecutorResults) realPresent(ctx context.Context, f flow.Flow) error {
+	parcel := h.Message.Parcel
+	logger := inslogger.FromContext(ctx)
+	msg := parcel.Message().(*message.ExecutorResults)
+
+	// now we have 2 different types of data in message.HandleExecutorResultsMessage
+	// one part of it is about consensus
+	// another one is about prepare state on new executor after pulse
+	// TODO make it in different goroutines
+
+	// prepare state after previous executor
+	procInitializeExecutionState := initializeExecutionState{
+		LR:  h.dep.lr,
+		msg: msg,
+	}
+	if err := f.Procedure(ctx, &procInitializeExecutionState, true); err != nil {
+		if err == flow.ErrCancelled {
+			return nil
+		}
+		err := errors.Wrap(err, "[ HandleExecutorResults ] Failed to initialize execution state")
+		return err
+	}
+
+	if !procInitializeExecutionState.Result.clarifyPending {
+		procClarifyPending := ClarifyPendingState{
+			es:     procInitializeExecutionState.Result.es,
+			parcel: parcel,
+		}
+		procClarifyPending.Dep.ArtifactManager = h.dep.lr.ArtifactManager
+
+		if err := f.Procedure(ctx, &procClarifyPending, true); err != nil {
+			if err == flow.ErrCancelled {
+				return nil
+			}
+
+			err := errors.Wrap(err, "[ HandleExecutorResults ] Failed to clarify pending")
+			return err
+		}
+	}
+
+	ref := msg.GetReference()
+	procStartQueueProcessorIfNeeded := StartQueueProcessorIfNeeded{
+		es:  procInitializeExecutionState.Result.es,
+		dep: h.dep,
+		ref: &ref,
+	}
+	if err := f.Handle(ctx, procStartQueueProcessorIfNeeded.Present); err != nil {
+		if err == flow.ErrCancelled {
+			return nil
+		}
+
+		logger.Warn("[ HandleExecutorResults ] StartQueueProcessorIfNeeded returns error: ", err)
+		return errors.Wrap(err, "[ HandleExecutorResults ] Failed to process queue")
```

Comment:
I believe our code stile forbids such kind of code. Handle an error or just return it, don't do both.

Reply:
copy past of previous code

Reply:
nope, my bad